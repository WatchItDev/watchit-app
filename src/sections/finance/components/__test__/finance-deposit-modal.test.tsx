import { describe, expect, it, vi } from "vitest";
import { render, screen, fireEvent } from "@testing-library/react";
import FinanceDepositModal from "../finance-deposit-modal";

vi.mock("@src/hooks/use-auth.ts", () => ({
  useAuth: vi.fn(() => ({
    session: {
      address: "0x123456789abcdef",
    },
  })),
}));

vi.mock("@src/hooks/protocol/use-deposit.ts", () => ({
  useDeposit: vi.fn(() => ({
    mutate: vi.fn(),
    isLoading: false,
  })),
}));

vi.mock("../finance-deposit", () => ({
  default: ({
    address,
    recipient,
    depositHook,
  }: {
    address: string;
    recipient: string;
    depositHook: {
      mutate: () => void;
      isLoading: boolean;
    };
  }) => (
    <div data-testid="FinanceDeposit">
      {address || "null"}-{recipient || "null"}-
      {depositHook && typeof depositHook.mutate === "function" ? "mockedDepositHook" : "noHook"}
    </div>
  ),
}));

vi.mock("@src/sections/finance/components/finance-deposit-from-stripe", () => ({
  default: () => <div>Stripe Deposit</div>,
}));

vi.mock("@src/sections/finance/components/finance-deposit-from-metamask", () => ({
  default: ({ onClose }: { onClose: () => void }) => (
    <div>
      Metamask Deposit
      <button onClick={onClose}>Close Metamask</button>
    </div>
  ),
}));

vi.mock("@src/sections/finance/components/finance-deposit-from-smart-account", () => ({
  default: ({ onClose }: { onClose: () => void }) => (
    <div>
      Smart Account Deposit
      <button onClick={onClose}>Close Smart</button>
    </div>
  ),
}));

const mockOnClose = vi.fn();
const mockOpen = true;

const renderComponent = () => {
  return render(<FinanceDepositModal open={mockOpen} onClose={mockOnClose} />);
};

describe("[COMPONENTS]: <FinanceDepositModal />", () => {
  it("to match snapshot", () => {
    renderComponent();
    expect(document.body).toMatchSnapshot();
  });

  it("renders modal title", () => {
    renderComponent();
    expect(screen.getByText("Deposit from")).toBeInTheDocument();
  });

  it("displays all deposit method tabs", () => {
    renderComponent();
    expect(screen.getByText("Stripe")).toBeInTheDocument();
    expect(screen.getByText("Metamask")).toBeInTheDocument();
    expect(screen.getByText("Smart Account")).toBeInTheDocument();
  });

  it("renders Metamask content when clicked", () => {
    renderComponent();
    fireEvent.click(screen.getByText("Metamask"));
    expect(screen.getByText("Metamask Deposit")).toBeInTheDocument();
  });

  it("renders Smart Account content when clicked", () => {
    renderComponent();
    fireEvent.click(screen.getByText("Smart Account"));
    expect(screen.getByText("Smart Account Deposit")).toBeInTheDocument();
  });

  
});
