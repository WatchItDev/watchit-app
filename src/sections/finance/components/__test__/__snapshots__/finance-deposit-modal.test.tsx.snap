// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`[COMPONENTS]: <FinanceDepositModal /> > to match snapshot 1`] = `
<body
  style="padding-right: 1024px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="MuiDialog-root MuiModal-root css-1lhei7z-MuiModal-root-MuiDialog-root"
    role="presentation"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root MuiModal-backdrop css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-testid="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
      role="presentation"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <div
        aria-labelledby=":r0:"
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthXs MuiDialog-paperFullWidth css-twia2z-MuiPaper-root-MuiDialog-paper"
        role="dialog"
      >
        <h2
          class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-1ss771v-MuiTypography-root-MuiDialogTitle-root"
          id=":r0:"
        >
          Deposit from
        </h2>
        <div
          class="MuiTabs-root css-rb36r6-MuiTabs-root"
        >
          <div
            class="MuiTabs-scrollableX MuiTabs-hideScrollbar css-oqr85h"
            style="width: 99px; height: 99px; position: absolute; top: -9999px; overflow: scroll;"
          />
          <div
            class="MuiTabs-scroller MuiTabs-hideScrollbar MuiTabs-scrollableX css-69z67c-MuiTabs-scroller"
            style="margin-bottom: 0px;"
          >
            <div
              class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
              role="tablist"
            >
              <button
                aria-selected="false"
                class="MuiButtonBase-root MuiTab-root MuiTab-labelIcon MuiTab-textColorPrimary css-aym9vq-MuiButtonBase-root-MuiTab-root"
                role="tab"
                tabindex="-1"
                type="button"
              >
                <span
                  class="MuiTab-iconWrapper"
                  data-icon="logos:stripe"
                  data-testid="iconify"
                />
                Stripe
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                aria-selected="false"
                class="MuiButtonBase-root MuiTab-root MuiTab-labelIcon MuiTab-textColorPrimary css-aym9vq-MuiButtonBase-root-MuiTab-root"
                role="tab"
                tabindex="-1"
                type="button"
              >
                <span
                  class="MuiTab-iconWrapper"
                  data-icon="logos:metamask-icon"
                  data-testid="iconify"
                />
                Metamask
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                aria-selected="true"
                class="MuiButtonBase-root MuiTab-root MuiTab-labelIcon MuiTab-textColorPrimary Mui-selected css-aym9vq-MuiButtonBase-root-MuiTab-root"
                role="tab"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiTab-iconWrapper"
                  data-icon="logos:ethereum-color"
                  data-testid="iconify"
                />
                Smart Account
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <span
              class="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
              style="left: 0px; width: 0px;"
            />
          </div>
        </div>
        <div>
          Smart Account Deposit
          <button>
            Close Smart
          </button>
        </div>
      </div>
    </div>
    <div
      data-testid="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;
